segway/gmtk/input_master.py

```diff
@@ -0,0 +1,627 @@
+        :param names: List[str]: names in this NameCollection
+        """
+        # Ensure all arguments are strings or lists of strings
+        if not all(isinstance(name, str) for name in names):
+            raise TypeError("All arguments must be str instances.")
+
+        super().__init__(names)
+
+    def __str__(self) -> str:
+        """
+        Returns string format of NameCollection object to be printed into the
+        input.master file (new lines to be added)
+        """
+        return "\n".join(list(self))
+
+    def get_header_info(self) -> str:
+        """
+        Return size of NameCollection, for header in input.master section.
+        """
+        return str(len(self))
+
+
+class OneLineKind(Array):
+    """
+    An abstract Python class acting as the parent for Python classes
+    representing Array-like GMTK kinds that have one-line string
+    representations in input.master.
+    It inherits the Array data structure for data storage and defines
+    the __str__ method for writing this data as one line.
+    """
+
+    def get_header_info(self) -> str:
+        """
+        Return string representation of own information, for header in
+        input.master section.
+        """
+        line = [str(len(self))]  # dimension
+        line.append(array2text(self))  # array values
+        return " ".join(line)
+
+
+class Mean(OneLineKind):
+    """
+    A single Mean object.
+    __init__ and __str__ methods defined in superclass.
+    """
+    kind = OBJ_KIND_MEAN
+
+
+class Covar(OneLineKind):
+    """
+    A single Covar object.
+    __init__ and __str__ methods defined in superclass.
+    """
+    kind = OBJ_KIND_COVAR
+
+
+# These types must be defined before they are referenced, so these constants
+# are defined here rather than the top of the file.
+
+# object classes that can support conversion
+CONVERTIBLE_CLASSES = {
+    OBJ_KIND_NAMECOLLECTION: NameCollection,
+    OBJ_KIND_MEAN: Mean,
+    OBJ_KIND_COVAR: Covar
+}
+
+# Types which can be auto-converted by convert
+ConvertableGMTKObjectType = Union[tuple(cls for cls in
+                                        CONVERTIBLE_CLASSES.values())]
+
+
+def convert(
+        cls: ConvertableGMTKObjectType,
+        value: Union[NumericArrayLike, ConvertableGMTKObjectType, List[str],
+                     str]
+) -> ConvertableGMTKObjectType:
+    """
+    Convert a provided attribute of input_master which a non-GMTK object into
+    a GMTK object of the provided class.
+    :param cls: GMTK class
+    :param value: Value to convert to GMTK object
+    """
+    # If provided value matches the desired class, no conversion needed
+    if isinstance(value, cls):
+        return value  # type: ignore[return-value]
+
+    # Otherwise, create new object of with provided data
+    return cls(value)
+
+
+class DPMF(OneLineKind):
+    """
+    A single DPMF object.
+    """
+    kind = OBJ_KIND_DPMF
+
+    # todo check if sums to 1.0
```

Comment:
-> `TODO` or `XXX`

Reply:
Please either post an issue or pull request for this.

Reply:
I'll resolve this change as part of the GMTK API integration branch.